portfolio/src/main/webapp/book-detail/book-detail.component.js

```diff
@@ -2,10 +2,11 @@ angular.
   module('bookDetail').
   component('bookDetail', {
     templateUrl: 'book-detail/book-detail.template.html',
-    controller: ['$routeParams',
-      function BookDetailController($routeParams) {
-          // This gets the ID from the URL. For example ...#!/book-detail/2  ->  2
-          this.bookId = $routeParams.bookId;
-      }
-    ]
+    controller: function BookDetailController($scope,$http,$routeParams) {
```

Comment:
Let's avoid explicitly passing $scope into the controller params. AngularJS kinda already does it for us behind the scenes and we can just set data like 'this.data' rather than '$scope.data'. 

This best practice (and short explanation) is here:
https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#style-y031

Once you do this one, go ahead and do the next one too (Y032) :P 
https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#style-y032

Reply:
Once you do those two, we might be able to go back and do what is recommended in Y030 in our views (html file).
https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#style-y031

I must admit that I'm reading about these best practices today, and that my code from yesterday doesn't quite abide by them! But easier to fix it all now than once we write more code!

Reply:
Also note: this isn't really an official Google style guide (just some good samaritan on Github), but I like the explanations and they seem consistent with what I'm seeing in the google style guide.